### ISSUE TITLE
Dispatch to `Integrate(Delta, ...)` in `normalize_integrate_contraction`

### ISSUE THREAD

**Issue by ordabayevy** [pyro-ppl/funsor#550]
There is a difference between how individual `Delta` terms are integrated in `normalize_integrate_contraction` and `Integrate` pattern for `Delta`. The former only performs `integrand` substitution (line 105) and the latter performs substitutions for both `integrand` (line 164) and `log_measure` (line 165). This can lead to differences when the `log_density` in `Delta` term is not zero. If the latter is more appropriate way then this probably can be easily fixed by dispatching:

```diff
    for delta in delta_terms:
+        args = delta, integrand, reduced_vars & delta.input_vars
+        integrand = eager.dispatch(Integrate, *args)(*args)
-        integrand = integrand(
-            **{
-                name: point
-                for name, (point, log_density) in delta.terms
-                if name in reduced_names.intersection(integrand.inputs)
-            }
        )
```

https://github.com/pyro-ppl/funsor/blob/e66bc6f6b5bc19d51e163666090bf22f6c0255d7/funsor/integrate.py#L90-L112

https://github.com/pyro-ppl/funsor/blob/e66bc6f6b5bc19d51e163666090bf22f6c0255d7/funsor/integrate.py#L153-L167

**Comment by eb8680** [comment]
I would think the latter is more correct, since the variables are bound and thus no longer available after the rule has been applied, but I'm not completely sure how this should work.

**Comment by ordabayevy** [comment]
Closed via #551
